src/app/common/hooks/useFilterState.ts

```diff
@@ -1,16 +1,17 @@
 import * as React from 'react';
 import { IFilterValues, FilterCategory } from '../components/FilterToolbar';
+import { IPlan } from '@app/queries/types';
 
 export interface IFilterStateHook<T> {
   filterValues: IFilterValues;
   setFilterValues: (values: IFilterValues) => void;
   filteredItems: T[];
 }
 
-export const useFilterState = <T>(
-  items: T[],
-  filterCategories: FilterCategory[]
-): IFilterStateHook<T> => {
+export const useFilterState = (
```

Comment:
This was fine with `<T>`, we just have to thread the `T` down through all the interfaces with generics at every level. `IPlan` should only be in the table.